buildSrc/src/main/groovy/com/uber/okbuck/core/model/java/JavaTarget.groovy

```diff
@@ -1,84 +1,74 @@
 package com.uber.okbuck.core.model.java
 
 import com.android.builder.model.LintOptions
+import com.google.common.base.Preconditions
 import com.uber.okbuck.OkBuckGradlePlugin
 import com.uber.okbuck.core.model.base.Scope
 import com.uber.okbuck.core.model.jvm.JvmTarget
 import com.uber.okbuck.core.util.LintUtil
-import com.uber.okbuck.core.util.ProjectUtil
 import org.gradle.api.JavaVersion
 import org.gradle.api.Project
 
 abstract class JavaTarget extends JvmTarget {
 
-    protected static final String TEST_PREFIX = "test"
+    protected static final String UNIT_TEST_PREFIX = "test"
 
-    private static final Set<String> JAVA_COMPILE_CONFIGS = ["compile"]
-    private static final Set<String> JAVA_APT_CONFIGS = ["apt"]
-    public static final Set<String> JAVA_PROVIDED_CONFIGS = ["compileOnly", "provided"]
+    public static final Set<String> JAVA_DEPS_CONFIGS = ["runtimeClasspath"]
```

Comment:
Lets use the constants from https://github.com/gradle/gradle/blob/master/subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java